File: 2025/46. Permutations/Solution2.java
class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        boolean[] used = new boolean[nums.length];
        findPermutation(nums, 0, used, new ArrayList<>(), res);
        return res;
    }

    public void findPermutation(int[] nums, int pos, boolean[] used, List<Integer> current, List<List<Integer>> res) {
        if (current.size() == nums.length) {
            res.add(new ArrayList<>(current));
            return;
        }
        for (int i = 0; i < nums.length; i++) {
            if (!used[i]) {
                used[i] = true;
                current.add(nums[i]);
                findPermutation(nums, pos + 1, used, current, res);
                current.remove(current.size() - 1);
                used[i] = false;
            }
        }
    }
}

File: 2025/2966. Divide Array Into Arrays With Max Difference/solution2.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        n = len(nums)
        if n % 3 != 0:
            return []
        nums = sorted(nums)
        res = []
        for i in range(0, n - 2, 3):
            if nums[i] + k < nums[i + 2]:
                return []
            res.append(nums[i:i+3])
        return res 